test/Spec.hs

```diff
@@ -0,0 +1,195 @@
+  , SC.testProperty "singleton" $ seeded $ Range.singleton @_ @a
+  , rangeSpec @a
+  -- TODO: Add more tests
+  ]
+
+floatingSpec ::
+     forall a.
+     (SC.Serial IO a, Typeable a, Num a, Ord a, Random a, UniformRange a, Show a)
+  => TestTree
+floatingSpec  =
+  testGroup ("(" ++ showsType @a ")")
+  [ SC.testProperty "uniformR" $ seeded $ Range.uniformRangeWithin @_ @a
+  -- TODO: Add more tests
+  ]
+
+runSpec :: TestTree
+runSpec = testGroup "runGenState_ and runPrimGenIO_"
+    [ SC.testProperty "equal outputs" $ seeded $ \g -> monadic $ Run.runsEqual g ]
+
+-- | Create a StdGen instance from an Int and pass it to the given function.
+seeded :: (StdGen -> a) -> Int -> a
+seeded f = f . mkStdGen
+
+
+instance Monad m => Serial m CFloat where
+  series = coerce <$> (series :: Series m Float)
+instance Monad m => Serial m CDouble where
+  series = coerce <$> (series :: Series m Double)
+instance Monad m => Serial m CBool where
+  series = coerce <$> (series :: Series m HTYPE_BOOL)
+instance Monad m => Serial m CChar where
+  series = coerce <$> (series :: Series m HTYPE_CHAR)
+instance Monad m => Serial m CSChar where
+  series = coerce <$> (series :: Series m HTYPE_SIGNED_CHAR)
+instance Monad m => Serial m CUChar where
+  series = coerce <$> (series :: Series m HTYPE_UNSIGNED_CHAR)
+instance Monad m => Serial m CShort where
+  series = coerce <$> (series :: Series m HTYPE_SHORT)
+instance Monad m => Serial m CUShort where
+  series = coerce <$> (series :: Series m HTYPE_UNSIGNED_SHORT)
+instance Monad m => Serial m CInt where
+  series = coerce <$> (series :: Series m HTYPE_INT)
+instance Monad m => Serial m CUInt where
+  series = coerce <$> (series :: Series m HTYPE_UNSIGNED_INT)
+instance Monad m => Serial m CLong where
+  series = coerce <$> (series :: Series m HTYPE_LONG)
+instance Monad m => Serial m CULong where
+  series = coerce <$> (series :: Series m HTYPE_UNSIGNED_LONG)
+instance Monad m => Serial m CPtrdiff where
+  series = coerce <$> (series :: Series m HTYPE_PTRDIFF_T)
+instance Monad m => Serial m CSize where
+  series = coerce <$> (series :: Series m HTYPE_SIZE_T)
+instance Monad m => Serial m CWchar where
+  series = coerce <$> (series :: Series m HTYPE_WCHAR_T)
+instance Monad m => Serial m CSigAtomic where
+  series = coerce <$> (series :: Series m HTYPE_SIG_ATOMIC_T)
+instance Monad m => Serial m CLLong where
+  series = coerce <$> (series :: Series m HTYPE_LONG_LONG)
+instance Monad m => Serial m CULLong where
+  series = coerce <$> (series :: Series m HTYPE_UNSIGNED_LONG_LONG)
+instance Monad m => Serial m CIntPtr where
+  series = coerce <$> (series :: Series m HTYPE_INTPTR_T)
+instance Monad m => Serial m CUIntPtr where
+  series = coerce <$> (series :: Series m HTYPE_UINTPTR_T)
+instance Monad m => Serial m CIntMax where
+  series = coerce <$> (series :: Series m HTYPE_INTMAX_T)
+instance Monad m => Serial m CUIntMax where
+  series = coerce <$> (series :: Series m HTYPE_UINTMAX_T)
```

Comment:
Could you possibly avoid `CPP` here? `Serial` instances for numeric types are pretty trivial, I believe one can put `series = fromInteger <$> series` for all `Foreign.C.Types.`

Reply:
`fromInteger <$> series` won't work without specifying the original type `a` in `fromIntegral :: a -> b`. In other words, CPP could only be avoided by specifying the actual type, which can change depending on the environment, making the code non-portable
Regardless of this, there is more CPP coming to the module in https://github.com/idontgetoutmuch/random/pull/147 so we can cut the losses against CPP